File: search-engine/src/main/java/com/memdb/service/kafka/CustomDeserializer.java
package com.memdb.service.kafka;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.memdb.service.kafka.dto.CaptionQueueDto;
import org.apache.kafka.common.serialization.Deserializer;

import java.io.IOException;
import java.io.UncheckedIOException;

public class CustomDeserializer implements Deserializer<Object> {
    @Override
    public Object deserialize(String topic, byte[] data) {
        ObjectMapper objectMapper = new ObjectMapper();
        try {
            return objectMapper.readValue(data, CaptionQueueDto.class);
        } catch (IOException e) {
            throw new UncheckedIOException("Deserialization Error", e);
        }
    }
}

File: tg-bot/src/main/kotlin/service/kafka/CustomDeserializer.kt
package service.kafka

import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.kafka.common.serialization.Deserializer
import service.kafka.dto.CaptionCallbackDto
import java.io.IOException
import java.io.UncheckedIOException


class CustomDeserializer : Deserializer<CaptionCallbackDto> {
    override fun deserialize(topic: String, data: ByteArray): CaptionCallbackDto {
        val objectMapper = ObjectMapper()
        try {
            return objectMapper.readValue(data, CaptionCallbackDto::class.java)
        } catch (e: IOException) {
            throw UncheckedIOException("Deserialization Error", e)
        }
    }
}

